corehq/toggles/__init__.py

```diff
@@ -1138,6 +1138,13 @@ def _ensure_valid_randomness(randomness):
     namespaces=[NAMESPACE_DOMAIN],
 )
 
+PAUSE_DATA_FORWARDING = StaticToggle(
+    'pause_data_forwarding',
+    "Pause all data forwarding from this domain",
+    TAG_INTERNAL,
+    namespaces=[NAMESPACE_DOMAIN],
+)
```

Comment:
I saw that `is_paused` is also used in the model manager method `all_ready`, but I don't actually see that method used anywhere other than tests.  I'd feel a little better if there were a clearer paradigm around `is_paused`, `next_attempt_at`, `is_ready`, and this feature flag (something like "everything should rout through `is_ready`").  I don't think that has to happen as part of this PR though, and it'd probably be best to get @kaapstorm 's take on how that should all fit together.

Reply:
The intention of `all_ready()` is to allow the repeat records queue to iterate repeaters instead of repeat records. Then we can (1) avoid overwhelming remote services, (2) make smarter decisions about whether we think the remote service is available, and (3) always send payloads in the order in which they were created.

Point 2 is especially relevant here. The way the repeat records queue works currently is that we have exponential back-off for individual payloads, but not for their endpoints. So if the endpoint is down, but we're sending 10,000 payloads to it, we're going to try all 10,000 payloads before we back off on the first payload. Not great!

We have intended to do this refactor for a long time. We have waited for the SQL migration to be completed. CommCare Analytics seems to be bumping the importance of this. I really would like it to happen soon.

Maybe once the refactor is done, this feature flag will no longer be necessary.

Reply:
Ah, are you saying that `all_ready` is prep work for an upcoming feature change?  I got confused when I didn't see any references to it outside of tests.

The pattern here certainly makes sense - I have no objections to that, I was more saying that it feels like there is a abstractable concept, "should we send repeat records from this repeater", that factors in whether the repeater is paused, whether this feature flag is enabled, and whether there's a `next_attempt_at` date in the future.  By checking these things individually in three different places, we risk answering the same question in different ways.  It looks like the concept of "ready" is meant to capture this, so I was saying it'd be nice to roll the feature flag into that abstraction rather than outside of it.

Reply:
I see, yeah I came across this and noticed it was only used in tests, so I didn't think much of it—but I was uneasy for the same reason as you Ethan, just the fact that this function clearly means to exclude anything that is (functionally) paused.

Reply:
Just to convince myself it's implementable, here's what adding that to this function would look like https://github.com/dimagi/commcare-hq/commit/5ae64250186bb5b92238bfaaa60e175337ceb30c. Decided not to include in this PR just to avoid it potentially delaying